test/backend/models.batch.test.js

```diff
@@ -109,6 +135,23 @@ describe('Report', function() {
     });
   });
 
+  it('should not give same reports to different users', function(done) {
```

Comment:
Does this test fail for the old code? I'm suspicious because it might only be a problem under high load.

Reply:
This test fails for the old code. It is an extreme case in where the ```Report.find``` in ```batch.lock``` actually returns the same ```ids``` for both users. The test fails as ```batch[0].length``` is 0: ```batch.load``` does not find any reports checked out by ```user```.  Only user2 gets the batch as the ```Report.update``` of ```user2``` overwrites the results of ```Report.update``` of user. 

In the high load scenario, where there are new reports coming in, there won't be such an extreme overlapping. Also, it can be that the ```batch.load``` of ```user``` can be executed before the ```Report.update``` of ```user2```, and, thus, the overwrite does not stop ```user``` from getting a batch, even if it is not checked out by ```user``` in the database anymore.

Reply:
:+1: